doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,456 @@
+
+.. rubric:: `encryption_udf.rsa_bit_threshold`
+
+These variable sets the threshold limits for the :ref:`asym-decrypt`, :ref:`asym-encrypt`, :ref:`asym-sign`, :ref:`asym-verify`, :ref:`create-priv-key`, and :ref:`create-pub-key`.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Description
+   * - command-line
+     - Yes
+   * - scope
+     - Global
+   * - data type
+     - unsigned integer
+   * - default
+     - 
+
+The range for this variable is 1024 to 10000. The value takes precedence over the OpenSSL maximum length value. 
+
+
+.. _function-examples:
+
+.. rubric:: Examples
+
+The following example sets the threshold variables, creates a private key, creates a public key, encrypts data, and decrypts data.
+
+
+.. code-block:: mysql
+
+    -- Set Global variable 
+    mysql> SET GLOBAL encryption_udf.dh_bits_threshold = 1024;
+
+.. code-block:: mysql
+ 
+    -- Create private key
+    mysql> SET @private_key = create_asymmetric_priv_key('RSA', 3072);
+
+    -- Create public key
+    mysql> SET @public_key = create_asymmetric_pub_key('RSA', 3072);
+
+    -- Encrypt data using the private key (you can also use the public key)
+    mysql> SET @ciphertext = asymmetric_encrypt('RSA', 'This text is secret', @private_key);
+
+    -- Decrypt data using the public key (you can also use the private key)
+    mysql> SET plaintext = asymmetric_decrypt('RSA', @ciphertext, @public_key);
+
+The following example generates a digest, generates a digest signature, and verifies the signature against the digest:
+
+.. code-block:: mysql
+
+   -- Generate a digest string
+   mysql> SET @digest = create_digest('SHA256', 'This is the text for digest');
+
+   -- Generate a digest signature 
+   mysql> SET @signature = asymmetric_sign('RSA', @digest, @private_key, 'SHA256');
+
+   -- Verify the signature against the digest
+   -- The @verify_signature must be equal to 1
+   mysql> SET @verify_signature = asymmetric_verify('RSA', @digest, @signature, @public_key, 'SHA256');
+
+The following examples create a symmetric keys using Diffie-Hellman secrets.
+
+.. code-block:: mysql
+
+   -- Generate a DH parameter
+   mysql> SET @dh-parameter = create_dh_parameters(3072);
+
+   -- Generate DH key pairs 
+   mysql> SET @private_1 = create_asymmetric_priv_key('DH', @dh_parameter);
+   mysql> SET @public_1 = create_asymmetric_pub_key('DH', @private_1);
+   mysql> SET @private_2 = create_asymmetric_priv_key('DH', @dh_parameter);
+   mysql> SET @public_2 = create_asymmetric_pub_key('DH', @private_2);
+
+  -- Generate a symmetric key using the public_1 and private_2
+  -- The @symetric_1 must be identical to @symetric_2
+  mysql> SET symmetric_1 = asymmetric_derive(@public_1, @private_2);
+
+  -- Generate a symmetric key using the public_2 and private_1
+    -- The @symetric_2 must be identical to @symetric_1
+  mysql> SET symmetric_2 = asymmetric_derive(@public_2, @private_1);
+
+The following example creates uses ``SET``, ``SELECT``, and ``INSERT``
```

Comment:
The following example creates uses ``SET``, ``SELECT``, and ``INSERT`` -> this sentence seems to be incomplete.

Reply:
Fixed.